source/Plugins/ExpressionParser/Swift/SwiftASTManipulator.cpp

```diff
@@ -814,7 +816,8 @@ void SwiftASTManipulator::InsertResult(
     SwiftASTManipulator::ResultLocationInfo &result_info) {
   swift::ASTContext &ast_context = m_source_file.getASTContext();
 
-  CompilerType return_ast_type(result_type.getPointer());
+  CompilerType return_ast_type(SwiftASTContext::GetSwiftASTContext(&ast_context),
+                               result_type.getPointer());
```

Comment:
This is dangerous, because it is not necessarily equivalent to the original code. Without having read the context, there is no guarantee that `ast_context` is equivalent to `result_type->getASTContext()`. There are many concurrent SwiftASTContexts floating around.

Reply:
I see. In that case, we should almost always be asking the swift::Type for its ASTContext when constructing a CompilerType then?

Reply:
If that's the case, then a wrapper or something would be a great idea. It would cut down on boilerplate significantly.